test/functional/wallet_conflicts.py

```diff
@@ -123,5 +129,148 @@ def run_test(self):
         assert_equal(former_conflicted["confirmations"], 1)
         assert_equal(former_conflicted["blockheight"], 217)
 
+    def test_mempool_conflict(self):
+        self.nodes[0].createwallet("alice")
+        alice = self.nodes[0].get_wallet_rpc("alice")
+
+        self.nodes[1].createwallet("bob")
+        bob = self.nodes[1].get_wallet_rpc("bob")
+
+        self.generatetoaddress(self.nodes[0], COINBASE_MATURITY + 3, alice.getnewaddress())
+
+        self.log.info("Test a scenario where a transaction has a mempool conflict")
+
+        unspents = [{"txid" : element["txid"], "vout" : element["vout"]} for element in alice.listunspent()]
+        assert_equal(len(unspents), 3)
+
+        raw_tx = alice.createrawtransaction(inputs=[unspents[0], unspents[1]], outputs=[{bob.getnewaddress() : 49.9999}])
+        tx1 = alice.signrawtransactionwithwallet(raw_tx)['hex']
+
+        raw_tx = alice.createrawtransaction(inputs=[unspents[1], unspents[2]], outputs=[{bob.getnewaddress() : 49.99}])
+        tx2 = alice.signrawtransactionwithwallet(raw_tx)['hex']
+
+        raw_tx = alice.createrawtransaction(inputs=[unspents[2]], outputs=[{bob.getnewaddress() : 24.9899}])
+        tx3 = alice.signrawtransactionwithwallet(raw_tx)['hex']
+
+        tx1 = alice.sendrawtransaction(tx1)
+
+        assert_equal(alice.listunspent()[0]["txid"], unspents[2]["txid"])
+        assert_equal(alice.getbalance(), 25)
+
+        tx2 = alice.sendrawtransaction(tx2)
+
+        # Check that the 0th unspent is now available because the transaction spending it has been replaced in the mempool
+        assert_equal(alice.listunspent()[0]["txid"], unspents[0]["txid"])
+        assert_equal(alice.getbalance(), 25)
+
+        assert_equal(alice.gettransaction(tx1)["mempoolconflicts"], [tx2])
+
+        self.log.info("Test scenario where a mempool conflict is removed")
+
+        tx3 = alice.sendrawtransaction(tx3)
+
+        assert_equal(alice.gettransaction(tx1)["mempoolconflicts"], [])
+
+        # now all of alice's outputs should be considered spent
+        assert_equal(alice.listunspent(), [])
+        assert_equal(alice.getbalance(), 0)
+
+        alice.sendrawtransaction(alice.gettransaction(tx1)['hex'])
+        self.generate(self.nodes[0], 3)
+        assert_equal(alice.getbalance(), 75)
+
+    def test_mempool_and_block_conflicts(self):
+        alice = self.nodes[0].get_wallet_rpc("alice")
+        bob = self.nodes[1].get_wallet_rpc("bob")
+
+        self.log.info("Test a scenario where a transaction has both a block conflict and a mempool conflict")
+        unspents = [{"txid" : element["txid"], "vout" : element["vout"]} for element in alice.listunspent()]
+
+        assert_equal(bob.getbalances()["mine"]["untrusted_pending"], 0)
+
+        self.disconnect_nodes(0, 1)
+
+        raw_tx = alice.createrawtransaction(inputs=[unspents[0]], outputs=[{bob.getnewaddress() : 24.99999}])
+        raw_tx1 = alice.signrawtransactionwithwallet(raw_tx)['hex']
```

Comment:
In 003efbbe45079c4416810a025b2bc372559dff15 "test: Add tests for wallet mempool conflicts"

This can be compressed into one line using the `send` RPC.

```suggestion
        raw_tx1 = alice.send(outputs=[{bob.getnewaddress(): 24.9999}], inputs=[unspents[0]], add_to_wallet=False)["hex"]
```

Reply:
These two things are not technically equivalent, because `send` creates a change output here, which results in intermittent failures for the rest of the test. Is there any way to tell `send` not to create a change output?